src/values/acceleration-structure.cxx

```diff
@@ -0,0 +1,1484 @@
+    bool rayFlagSkipTriangles = false;
+    bool rayFlagSkipAABBs = false;
+
+    void setFlags() {
+        rayFlagNone = (rayFlags | spv::RayFlagsMask::RayFlagsMaskNone) == 0; // TODO: reason to have this?
+        rayFlagOpaque = rayFlags & spv::RayFlagsMask::RayFlagsOpaqueKHRMask;
+        rayFlagNoOpaque = rayFlags & spv::RayFlagsMask::RayFlagsNoOpaqueKHRMask;
+        rayFlagTerminateOnFirstHit = rayFlags & spv::RayFlagsMask::RayFlagsTerminateOnFirstHitKHRMask;
+        rayFlagSkipClosestHitShader = rayFlags & spv::RayFlagsMask::RayFlagsSkipClosestHitShaderKHRMask;
+        rayFlagCullBackFacingTriangles = rayFlags & spv::RayFlagsMask::RayFlagsCullBackFacingTrianglesKHRMask;
+        rayFlagCullFrontFacingTriangles = rayFlags & spv::RayFlagsMask::RayFlagsCullFrontFacingTrianglesKHRMask;
+        rayFlagCullOpaque = rayFlags & spv::RayFlagsMask::RayFlagsCullOpaqueKHRMask;
+        rayFlagCullNoOpaque = rayFlags & spv::RayFlagsMask::RayFlagsCullNoOpaqueKHRMask;
+        rayFlagSkipTriangles = rayFlags & spv::RayFlagsMask::RayFlagsSkipTrianglesKHRMask;
+        rayFlagSkipAABBs = rayFlags & spv::RayFlagsMask::RayFlagsSkipAABBsKHRMask; // skip procedurals
+    }
+
+    void clearTrace() {
+        while (!nodesToEval.empty()) {
+            nodesToEval.pop();
+        }
+    }
+
+    void initTrace() {
+        closestInstance = nullptr;
+        closestPrimitiveID = 0;
+        committedIntersectionType = CommittedIntersectionType::None;
+        candidateIntersectionType = CandidateIntersectionType::Invalid;
+        activeTrace = true;
+        nodesToEval.push(&root);
+    }
+
+public: // TODO: also traversal stuff
+
+    void resetTrace() {
+        activeTrace = false;
+        clearTrace();
+        initTrace();
+    }
+    
+    void initTrace(const unsigned& rayFlags,
+            const unsigned& cullMask,
+            const std::vector<float>& rayOrigin,
+            const std::vector<float>& rayDirection,            
+            const float& rayTMin,
+            const float& rayTMax,
+            const bool& useSBT,
+            const unsigned offsetSBT = 0,
+            const unsigned strideSBT = 0,
+            const unsigned missIndex = 0) {
+        
+        assert(rayOrigin.size() == 3 && rayDirection.size() == 3);
+        glm::vec4 convertedRayOrigin = glm::make_vec4(rayOrigin.data());
+        convertedRayOrigin.w = 1.0f;
+        glm::vec4 convertedRayDirection = glm::make_vec4(rayDirection.data());
+        convertedRayDirection.w = 0.0f;
+
+        this->rayFlags = rayFlags;
+        this->cullMask = cullMask;
+        this->rayOrigin = convertedRayOrigin;
+        this->rayDirection = convertedRayDirection;
+        this->rayTMin = rayTMin;
+        this->rayTMax = rayTMax;
+
+        this->useSBT = useSBT;
+        this->offsetSBT = offsetSBT;
+        this->strideSBT = strideSBT;
+        this->missIndex = missIndex;
+
+        setFlags();
+        resetTrace();
+    }
+
+    void initTrace(const unsigned& rayFlags,
```

Comment:
This method does not follow the established style. Please edit the formatting to match what you see in instruction.cxx (for example, look at Instruction::readOp). Thanks.

Reply:
Should add this clang format option, `AlignAfterOpenBracket: BlockIndent`, to `.clang-format`?

Reply:
Give it a try. I don't have a ton of experience with the clang-format, so I am still tweaking it to match the code style I expect.